beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -814,7 +814,11 @@ impl<T: BeaconChainTypes> IntoFullyVerifiedBlock<T> for SignedBeaconBlock<T::Eth
         self,
         chain: &BeaconChain<T>,
     ) -> Result<FullyVerifiedBlock<T>, BlockSlashInfo<BlockError<T::EthSpec>>> {
-        SignatureVerifiedBlock::check_slashable(self, chain)?
+        // Perform an early check to prevent wasting time on irrelevant blocks.
+        let block_root = check_block_relevancy(&self, None, chain)
+            .map_err(|e| BlockSlashInfo::SignatureValid(self.signed_block_header(), e))?;
```

Comment:
```suggestion
            .map_err(|e| BlockSlashInfo::SignatureNotChecked(self.signed_block_header(), e))?;
```

Although duplicate blocks are valid and have valid signatures, we need to return `SignatureNotChecked` here so that the slasher can check sigs on blocks that fail for other reasons (e.g. future slot, older than finalization, etc).